src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketBuildStatusClientImplTest.java

```diff
@@ -0,0 +1,65 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.http.RetryOnRateLimitConfig;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import com.atlassian.bitbucket.jenkins.internal.model.BuildState;
+import com.atlassian.bitbucket.jenkins.internal.model.TestResults;
+import okhttp3.HttpUrl;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.annotation.Nullable;
+
+import java.util.function.Consumer;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BitbucketBuildStatusClientImplTest {
+
+    @Mock
+    BitbucketRequestExecutor bitbucketRequestExecutor;
+
+    String revisionSha = "revisionSha";
+
+    BitbucketBuildStatusClientImpl bitbucketBuildStatusClient;
+
+    @Before
+    public void setup() {
+        when(bitbucketRequestExecutor.getBaseUrl()).thenReturn(HttpUrl.parse("http://example.com"));
+        bitbucketBuildStatusClient = new BitbucketBuildStatusClientImpl(bitbucketRequestExecutor, revisionSha);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testPost() {
+        BitbucketBuildStatus.Builder buildStatusBuilder = createTestBuildStatus("ref");
+        BitbucketBuildStatus buildStatus = buildStatusBuilder.legacy().build();
```

Comment:
nit: I don't think you need to specify the legacy here if it's going to be done in the `BitbucketBuildStatusClient`